api/handlers.go

```diff
@@ -212,12 +235,45 @@ func (api *API) userGET(w http.ResponseWriter, req *http.Request, _ httprouter.P
 
 // userLimitsGET returns the speed limits which apply to this user.
 func (api *API) userLimitsGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	u := api.userFromRequest(req)
-	if u == nil || u.QuotaExceeded {
+	t, err := tokenFromRequest(req)
+	if err != nil {
 		api.WriteJSON(w, database.UserLimits[database.TierAnonymous])
 		return
 	}
-	api.WriteJSON(w, database.UserLimits[u.Tier])
+	tk, err := jwt.ValidateToken(t)
+	if err != nil {
+		api.WriteJSON(w, database.UserLimits[database.TierAnonymous])
+		return
+	}
+	s, exists := tk.Get("sub")
+	if !exists {
+		api.staticLogger.Tracef("Token without a sub.")
+		api.WriteJSON(w, database.TierAnonymous)
+		return
+	}
+	sub := s.(string)
+	// If the user is not cached, or they were cached too long ago we'll fetch
+	// their data from the DB.
+	if c, exists := userLimitCache[sub]; !exists || c.LastUpdate.Before(time.Now().Add(-userLimitCacheTTL).UTC()) {
```

Comment:
```suggestion
	if c, exists := userLimitCache[sub]; !exists || c.LastUpdate.Since(time.Now()) > userLimitCacheTTL {
```

Reply:
It turns out it's `c.LastUpdate.Sub(time.Now()) > userLimitCacheTTL`